setup.py

```diff
@@ -93,12 +98,34 @@ def run(self):
             )
 
 
+def get_version(version: "ScmVersion") -> str:
+    from setuptools_scm import version as scm_version
+
+    if version.node:
+        version.node = str(int(version.node.lstrip("g"), 16))
+    if version.exact:
+        return version.format_with("{tag}")
+    else:
+        return version.format_next_version(
+            scm_version.guess_next_version,
+            fmt="{guessed}.dev{node}",
+        )
+
+
+def get_local_version(version: "ScmVersion", time_format="%Y%m%d") -> str:
```

Comment:
Add docstring explaining what this does. (Was non-obvious to me without looking up format_choice; I see this is something added if working directory is dirty.)

Reply:
Apologies, will document. Spent so long reading the docs for this thing that forgot other people don't actually know how this works 😂